sherpa/astro/ui/tests/test_astro_ui_plot.py

```diff
@@ -3734,17 +3734,74 @@ def test_set_plot_opt_with_plot_y():
 
 @pytest.mark.parametrize("cls",
                          [sherpa.ui.utils.Session, sherpa.astro.ui.utils.Session])
-def test_set_opt_invalid(cls):
-    """Check we error out if called with an invalid option"""
+@pytest.mark.parametrize("name", [None, 1, {}])
+def test_set_opt_not_string(cls, name):
+    """Check we error out if called with an argument that is not a string."""
+
+    s = cls()
+    with pytest.raises(ArgumentTypeErr) as exc:
+        s.set_ylog(name)
+
+    assert str(exc.value) == "'plottype' must be a string"
+
+
+@pytest.mark.parametrize("cls",
+                         [sherpa.ui.utils.Session, sherpa.astro.ui.utils.Session])
+@pytest.mark.parametrize("name", ["notdata",  "fit-allthe-things", " fi", "fi",
+                                  "source_component", "model_component",
+                                  "astrodata", "astrosource", "astromodel",
+                                  "flux",
+                                  "bkg_model", "bkg_source", "bkg_fit", "bkg_resid", "bkg_ratio", "bkg_chisqr", "bkg_delchi"])
+def test_set_opt_invalid(cls, name):
+    """Check we error out if called with an invalid option.
+
+    The logic for what is and isn't supported in calls like set_xlog
+    has not really been documented, so just test smoe possible names
+    which do not work. This is a regression test, so values may be
+    added or removed over time.
+    """
 
     s = cls()
     with pytest.raises(PlotErr) as exc:
-        s.set_xlog('notdata')
+        s.set_xlog(name)
 
-    msg = "Plot type 'notdata' not found in ["
+    msg = f"Plot type '{str(name).strip()}' not found in ["
     assert str(exc.value).startswith(msg)
 
 
+@pytest.mark.parametrize("cls",
+                         [sherpa.ui.utils.Session, sherpa.astro.ui.utils.Session])
+@pytest.mark.parametrize("name", ["all", " all", "all ", "  all   ",
+                                  "data", " data", "data ", " data   ",
+                                  "source", "model",
+                                  "resid", "ratio", "delchi", "chisqr",
+                                  "fit",
+                                  "compmodel", "compsource"])
+def test_set_opt_valid(cls, name):
+    """What names are accepted for set_xlog/ylog/...?
+
+    The logic for what is and isn't supported in calls like set_xlog
+    has not really been documented, so just test smoe possible names
+    which do work. This is a regression test, so values may be
+    added or removed over time.
+    """
+
+    # All we care about here is that the call does not raise an error
+    s = cls()
+    s.set_ylinear(name)
```

Comment:
I suggest to text one x and one log, too, so can be done like this. If we go through that many strings, then we can affard this, too!
```suggestion
    s.set_ylinear(name)
    s.set_xlog(name)
```

Reply:
We do have a bunch of other tests that check these routines work, and given how they are written they "should" work, but I can add them as it doesn't really slow things down. WIll do it in a rebase (for reference, the only relevant commands are

    set_xlog
    set_ylog
    set_xlinear
    set_ylinear